pkg/kubelet/kubelet.go

```diff
@@ -1493,7 +1491,25 @@ func (kl *Kubelet) initializeRuntimeDependentModules() {
 func (kl *Kubelet) Run(updates <-chan kubetypes.PodUpdate) {
 	ctx := context.Background()
 	if kl.logServer == nil {
-		kl.logServer = http.StripPrefix("/logs/", http.FileServer(http.Dir("/var/log/")))
+		file := http.FileServer(http.Dir("/var/log/"))
+		if utilfeature.DefaultFeatureGate.Enabled(features.NodeLogQuery) {
+			kl.logServer = http.StripPrefix("/logs/", http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+				query := req.URL.Query()
+				if errs := validateNodeLogQueryCombination(query); len(errs) > 0 {
+					http.Error(w, errs.ToAggregate().Error(), http.StatusNotAcceptable)
+					return
+				}
+				services := parseQueryForServices(query["query"])
+				if len(services) > 0 {
+					journal.ServeHTTP(w, req)
+					return
+				}
```

Comment:
all the query params this will honor should be surfaced to InstallSystemLogHandler so it can include the names and documentation for those params in the registered routes

Reply:
Done though I am not 100% sure if I got it right. PTAL.